openassessment/xblock/static/js/spec/lms/oa_staff_area.js

```diff
@@ -0,0 +1,302 @@
+        });
+    });
+
+    describe('Staff Toolbar', function() {
+        beforeEach(function() {
+            loadFixtures('oa_base_course_staff.html');
+        });
+
+        var getVisibleStaffPanels = function(view) {
+            return $('.wrapper--ui-staff', view.element).not('.is--hidden');
+        };
+
+        it('shows the correct buttons with no panels initially', function() {
+            var view = createStaffArea(),
+                $buttons = $('.ui-staff__button', view.element);
+            expect($buttons.length).toBe(2);
+            expect($($buttons[0]).text().trim()).toEqual('Staff Tools');
+            expect($($buttons[1]).text().trim()).toEqual('Staff Info');
+            expect(getVisibleStaffPanels(view).length).toBe(0);
+        });
+
+        it('shows the "Staff Tools" panel when the button is clicked', function() {
+            var view = createStaffArea(),
+                $staffToolsButton = $('.button-staff-tools', view.element),
+                $visiblePanels;
+            expect($staffToolsButton.length).toBe(1);
+            expect($staffToolsButton).not.toHaveClass('is--active');
+            $staffToolsButton[0].click();
+            expect($staffToolsButton).toHaveClass('is--active');
+            $visiblePanels = getVisibleStaffPanels(view);
+            expect($visiblePanels.length).toBe(1);
+            expect($visiblePanels.first()).toHaveClass('wrapper--staff-tools');
+        });
+
+        it('shows the "Staff Info" panel when the button is clicked', function() {
+            var view = createStaffArea(),
+                $staffInfoButton = $('.button-staff-info', view.element),
+                $visiblePanels;
+            expect($staffInfoButton.length).toBe(1);
+            expect($staffInfoButton).not.toHaveClass('is--active');
+            $staffInfoButton[0].click();
+            expect($staffInfoButton).toHaveClass('is--active');
+            $visiblePanels = getVisibleStaffPanels(view);
+            expect($visiblePanels.length).toBe(1);
+            expect($visiblePanels.first()).toHaveClass('wrapper--staff-info');
+        });
+
+        it('hides the "Staff Tools" panel when the button is clicked twice', function() {
+            var view = createStaffArea(),
+                $staffToolsButton = $('.button-staff-tools', view.element);
+            expect($staffToolsButton.length).toBe(1);
+            expect($staffToolsButton).not.toHaveClass('is--active');
+            $staffToolsButton[0].click();
+            expect($staffToolsButton).toHaveClass('is--active');
+            $staffToolsButton[0].click();
+            expect($staffToolsButton).not.toHaveClass('is--active');
+        });
+    });
+
+    describe('Staff Tools', function() {
+        beforeEach(function() {
+            loadFixtures('oa_base_course_staff.html');
+        });
+
+        it('hides the "Staff Tools" panel when the close button is clicked', function() {
```

Comment:
Same note about code duplication-- as far as I can tell, the only difference between this test point and the one at line 290 is the actual button being clicked.